ncolony/ctllib.py

```diff
@@ -53,23 +53,31 @@ def _dumps(stuff):
 
 
 # pylint: disable=too-many-arguments,too-many-locals
-def add(places, name, cmd, args, env=None, uid=None, gid=None, extras=None,
-        env_inherit=None):
+def add(places, name, cmd, args,
+        env=None, uid=None, gid=None, extras=None,
+        env_inherit=None, _cmd_and_args=None):
     """Add a process.
 
     :param places: a Places instance
     :param name: string, the logical name of the process
     :param cmd: string, executable
     :param args: list of strings, command-line arguments
-    :param env: dictionary mapping strings to strings
-         (will be environment in subprocess)
+    :param env: dictionary mapping strings to strings (will be
+        environment in subprocess)
     :param uid: integer, uid to run the new process as
     :param gid: integer, gid to run the new process as
     :param extras: a dictionary with additional parameters
     :param env_inherit: a list of environment variables to inherit
+
     :returns: None
     """
-    args = [cmd]+args
+    if not (cmd or _cmd_and_args):
```

Comment:
`argparse` disallows the following:

* `python -m ncolony ctl add <name> --cmd command positional_command` 
* `python -m ncolony ctl add <name> --arg argument positional_command`

But it does allow:
* `python -m ncolony ctl add <name> --arg argument`

So I think this is sufficient.